rdfunit-junit/src/main/java/org/aksw/rdfunit/junit/RdfUnitJunitStatusTestExecutor.java

```diff
@@ -0,0 +1,36 @@
+package org.aksw.rdfunit.junit;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collection;
+
+import org.aksw.rdfunit.exceptions.TestCaseExecutionException;
+import org.aksw.rdfunit.io.reader.RDFModelReader;
+import org.aksw.rdfunit.sources.TestSource;
+import org.aksw.rdfunit.sources.TestSourceBuilder;
+import org.aksw.rdfunit.tests.executors.RLOGTestExecutor;
+import org.aksw.rdfunit.tests.query_generation.QueryGenerationSelectFactory;
+import org.aksw.rdfunit.tests.results.TestCaseResult;
+
+final class RdfUnitJunitStatusTestExecutor extends RLOGTestExecutor {
+
+    public RdfUnitJunitStatusTestExecutor() {
+        super(new QueryGenerationSelectFactory());
+    }
+
+    Collection<TestCaseResult> runTest(RdfUnitJunitTestCase rdfUnitJunitTestCase)
+            throws IllegalAccessException, InvocationTargetException {
+        final TestSource modelSource = new TestSourceBuilder()
+                .setPrefixUri("custom", "rdfunit")
```

Comment:
To get rid off hardcoding stuff we should go with an approach similar to
```

@RunWith(RdfUnitJunitRunner.class)
public class ExampleTest {

    @Schema
    public SchemaSource schemaSource =
SchemaSourceFactory.createSchemaSourceSimple("foaf",
Constants.FOAF_ONTOLOGY_URI,
RDFReaderFactory.createResourceReader(Constants.FOAF_ONTOLOGY_URI));


    @TestInput
    public TestSource testSource = new TestSourceBuilder()
            .setImMemSingle()
            .setInMemReader(RDFReaderFactory.createResourceReader("inputmodels/foaf.rdf"))
            .setReferenceSchemata(schemaSource)
            .build();
}

```

and we can define some factory methods to simplify object construction


On Wed, Sep 23, 2015 at 12:24 PM, Daniel Hiller <notifications@github.com>
wrote:

> In
> rdfunit-junit/src/main/java/org/aksw/rdfunit/junit/RdfUnitJunitStatusTestExecutor.java
> <https://github.com/AKSW/RDFUnit/pull/51#discussion_r40192614>:
>
> > +import org.aksw.rdfunit.sources.TestSource;
> > +import org.aksw.rdfunit.sources.TestSourceBuilder;
> > +import org.aksw.rdfunit.tests.executors.RLOGTestExecutor;
> > +import org.aksw.rdfunit.tests.query_generation.QueryGenerationSelectFactory;
> > +import org.aksw.rdfunit.tests.results.TestCaseResult;
> > +
> > +final class RdfUnitJunitStatusTestExecutor extends RLOGTestExecutor {
> > +
> > +    public RdfUnitJunitStatusTestExecutor() {
> > +        super(new QueryGenerationSelectFactory());
> > +    }
> > +
> > +    Collection<TestCaseResult> runTest(RdfUnitJunitTestCase rdfUnitJunitTestCase)
> > +            throws IllegalAccessException, InvocationTargetException {
> > +        final TestSource modelSource = new TestSourceBuilder()
> > +                .setPrefixUri("custom", "rdfunit")
>
> OK, so what would you propose regarding the setPrefixUri() call?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/AKSW/RDFUnit/pull/51/files#r40192614>.
>



-- 
Kontokostas Dimitris